homeassistant/components/camera/__init__.py

```diff
@@ -460,7 +464,11 @@ def __init__(self) -> None:
         self._warned_old_signature = False
         self.async_update_token()
         self._create_stream_lock: asyncio.Lock | None = None
-        self._webrtc_providers: list[CameraWebRTCProvider] = []
+        self._webrtc_provider: CameraWebRTCProvider | None = None
+        self._legacy_webrtc_provider: CameraWebRTCLegacyProvider | None = None
```

Comment:
I think we should remove these APIs rather than needing to support the complexity of these shims given there is no real usage -- Sent https://github.com/home-assistant/core/pull/128790

Reply:
As `rtsp_to_webrtc` has no support for async webRTC as there is currently no way to provide async candidate I suggest to merge https://github.com/home-assistant/core/pull/128790 after this PR

Reply:
I'm not sure I understand why it can't it just return the answer using that same API?

Reply:
(The tests pass, don't they? it seems to work fine?)

Reply:
The problem is not the answer.
The problem is that the offer will not include (all) candidates (it's even possible that the offer includes no candidates et all)

The candidates are received from the frontend async, and on the camera entity for each candidate, `async_on_webrtc_candidate` will be called. `rtsp_to_webrtc` has currently no support for it and is expecting that the offer includes all candidates

Reply:
Sure though won't sync still be supported generally e.g. for nest?